datasync/downloader/downloader.go

```diff
@@ -316,12 +316,20 @@ func (d *Downloader) UnregisterPeer(id string) error {
 func (d *Downloader) Synchronise(id string, head common.Hash, td *big.Int, mode SyncMode) error {
 	err := d.synchronise(id, head, td, mode)
 	switch err {
-	case nil:
-	case errBusy, errCanceled:
+	case nil, errBusy, errCanceled:
+		return err
+	}
+
+	if errors.Is(err, errInvalidChain) {
+		logger.Warn("Synchronisation failed, dropping peer", "peer", id, "err", err)
+		d.dropPeer(id)
+		return err
+	}
```

Comment:
Why did you omit the `nil` handling for `d.dropPeer`?

Reply:
Maybe because of the missing command `copydb` in Klaytn?

Reply:
I think even if we don't have that case, `nil` case check is useful.
@jeongkyun-oh Let's add it.

Reply:
@aidan-kwon Yes, `d.dropPeer` is nil only if `--copydb` flag is on. There is no `nil` checking for the same logic already.
@KimKyungup I will apply them to other places as well if necessary for consistency on another PR.

Reply:
Yes. Please apply them for consistency.